### ISSUE TITLE
Formatting rules are reported at wrong line (e.g. MaximumLineLength)

### ISSUE THREAD

**Issue by sschuberth** [arturbosch/detekt#1843]
## Expected Behavior

I expect *MaximumLineLength* to be reported at the line it occurred.

## Observed Behavior

*MaximumLineLength* is reported at `<correct line> - 2` (in my case).

## Steps to Reproduce

- Clone https://github.com/heremaps/oss-review-toolkit and checkout the `detekt-issue-1843` branch.
- Run `./gradlew :scanner:detekt`.
- Observe:

```
Ruleset: comments
Ruleset: complexity
Ruleset: empty-blocks
Ruleset: exceptions
Ruleset: formatting - 5min debt
        MaximumLineLength - [PsiWhiteSpace] at C:\Dev\oss-review-toolkit\scanner\src\funTest\kotlin\storages\PostgresStorageTest.kt:39:65
Ruleset: naming
Ruleset: performance
Ruleset: potential-bugs
Ruleset: style - 5min debt
        MaxLineLength - [createStorage] at C:\Dev\oss-review-toolkit\scanner\src\funTest\kotlin\storages\PostgresStorageTest.kt:41:5
```

Note that the ktlint-based *MaximumLineLength* is reported at line 39 (wrong), but the detekt-based *MaxLineLength* is reported at line 41 (correct) for the same file.

Also, running stand-alone ktlint 0.34.2 correctly reports 41, so the issue is with detekt, not with ktlint.

## Context

Might be related to https://github.com/arturbosch/detekt/issues/1688.

## Your Environment

* Version of detekt used: 1.0.0
* Version of Gradle used (if applicable): 5.6.0
* Operating System and version: Windows 10 x64
* Link to your project (if it's a public repository): https://github.com/heremaps/oss-review-toolkit/tree/detekt-issue-1843

**Comment by sschuberth** [comment]
Still present in detekt 1.0.1.

**Comment by arturbosch** [comment]
All detekt-formatting rules are reported with slightly wrong positions - #2120.

**Comment by sschuberth** [comment]
Oh! Should detekt then maybe say "*approximate* line X" for these rules to avoid confusion?

**Comment by arturbosch** [comment]
> Oh! Should detekt then maybe say "_approximate_ line X" for these rules to avoid confusion?

Hm, I think as most users use the autoCorrect for this it's not that big of an issue.
However we need to debug why we report them wrongly.
For detekt-rules we use the kotlin compilers method to transform line endings. KtLint uses own method to do this. For detekt-formatting we combine both methods which I think is the reason for this.
